### PR TITLE
docs(plugin) add refresh_token_ttl to oauth2 plugin

### PR DESCRIPTION
**Pull request by bob983**
### OAuth2 plugin documentation update

For Kong/kong#2942 - oauth2: Add custom TTL to refresh tokens

### CODE REVIEW

File: app/plugins/oauth2-authentication.md [hunk 1]
```diff
@@ -79,6 +79,7 @@ form parameter                                    | default | description
 `config.accept_http_if_already_terminated`<br>*optional* | `false` | Accepts HTTPs requests that have already been terminated by a proxy or load balancer and the `x-forwarded-proto: https` header has been added to the request. Only enable this option if the Kong server cannot be publicly accessed and the only entry-point is such proxy or load balancer.
 `config.anonymous`<br>*optional*           | `` | An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`
 `config.global_credentials`<br>*optional*   | `false` | An optional boolean value that allows to use the same OAuth credentials generated by the plugin with any other API whose OAuth 2.0 plugin configuration also has `config.global_credentials=true`.
+`config.refresh_token_ttl`<br>*optional*   | `1209600` | An optional integer value telling the plugin how many seconds a token/refresh token pair is stored before it's removed. Default value is 2 weeks. Set to `0` to keep the token/refresh token pair forever.
```
Comment (Tieske) [inline app/plugins/oauth2-authentication.md]: I think the terminology "before it's being removed" refers to the technical implementation, not the functional effect. So something like "expiration" or "validity" might be a better terminology here.

Comment (bob983) [reply app/plugins/oauth2-authentication.md]: Gotcha, let me reword it.

**Comment by bob983** [comment]
What about now?

File: app/plugins/oauth2-authentication.md [hunk 2]
```diff
@@ -79,6 +79,7 @@ form parameter                                    | default | description
 `config.accept_http_if_already_terminated`<br>*optional* | `false` | Accepts HTTPs requests that have already been terminated by a proxy or load balancer and the `x-forwarded-proto: https` header has been added to the request. Only enable this option if the Kong server cannot be publicly accessed and the only entry-point is such proxy or load balancer.
 `config.anonymous`<br>*optional*           | `` | An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`
 `config.global_credentials`<br>*optional*   | `false` | An optional boolean value that allows to use the same OAuth credentials generated by the plugin with any other API whose OAuth 2.0 plugin configuration also has `config.global_credentials=true`.
+`config.refresh_token_ttl`<br>*optional*   | `1209600` | An optional integer value telling the plugin how many seconds a token/refresh token pair is valid can be used to generate a new access token. Default value is 2 weeks. Set to `0` to keep the token/refresh token pair forever.
```
Comment (Tieske) [inline app/plugins/oauth2-authentication.md]: ...is valid _and_ can be...

Comment (bob983) [reply app/plugins/oauth2-authentication.md]: Oh man, can I even read? 😞  Thanks

**Comment by thibaultcha** [comment]
@bob983 I have changed the base of this PR to be `release/0.12`. Would you mind rebasing on top of that branch? Thank you!

**Comment by bob983** [comment]
Rebased and added the missing _and_.

File: app/plugins/oauth2-authentication.md [hunk 3]
```diff
@@ -79,6 +79,7 @@ form parameter                                    | default | description
 `config.accept_http_if_already_terminated`<br>*optional* | `false` | Accepts HTTPs requests that have already been terminated by a proxy or load balancer and the `x-forwarded-proto: https` header has been added to the request. Only enable this option if the Kong server cannot be publicly accessed and the only entry-point is such proxy or load balancer.
 `config.anonymous`<br>*optional*           | `` | An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`
 `config.global_credentials`<br>*optional*   | `false` | An optional boolean value that allows to use the same OAuth credentials generated by the plugin with any other API whose OAuth 2.0 plugin configuration also has `config.global_credentials=true`.
+`config.refresh_token_ttl`<br>*optional*   | `1209600` | An optional integer value telling the plugin how many seconds a token/refresh token pair is valid and can be used to generate a new access token. Default value is 2 weeks. Set to `0` to keep the token/refresh token pair forever.
```
Comment (thibaultcha) [inline app/plugins/oauth2-authentication.md]: We should also add a note that says this value was introduced in 0.12.0. Sadly, our plugins documentatio nis not versioned (yet) 😢

Comment (bob983) [reply app/plugins/oauth2-authentication.md]: OK, let me fix that :)

**Comment by bob983** [comment]
@thibaultcha fixed wording and added warning with Kong version info (stolen from `key-authentication` plugin)

**Comment by thibaultcha** [comment]
Just merged this manually to `release/0.12.0rc1` - it is finally getting out! Thank you for providing the documentation to this contribution :+1: 

(See https://github.com/Kong/kong/blob/release/0.12.0rc1/CHANGELOG.md for details on 0.12.0rc1)

**Pull request closed without merging**